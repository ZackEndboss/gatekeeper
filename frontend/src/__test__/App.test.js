import React from 'react';
import renderer from 'react-test-renderer';
import {MockStore} from './mockStore';
import {Provider} from 'react-redux';
import ConnectedApp, {App} from '../App';

jest.mock('../Header', () => 'Header');
jest.mock('../Toolbar', () => 'Toolbar');
jest.mock('../EditServiceDialog', () => 'EditServiceDialog');
jest.mock('../Group', () => 'Group');

describe('App component', () => {
    it('render without groupId', () => {
        const component = renderer.create(<App groups={['group1', 'group2']}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('render without state', () => {
        const component = renderer.create(<App groups={[]}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('render group1', () => {
        const component = renderer.create(<App groups={['group1', 'group2']}
                                               match={{params: {groupId: 'group1'}}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('render with non existing group', () => {
        const component = renderer.create(<App groups={['group1', 'group2']}
                                               match={{params: {groupId: 'some_group'}}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('load from state', () => {
    it('should load the state correctly', () => {
        const storeMock = MockStore({
            'gates': {
                'ftx': {},
                'fty': {}
            }
        });

        const subject = (
            <Provider store={storeMock}>
                <ConnectedApp match={{params: {groupId: 'ftx'}}}/>
            </Provider>);
        const component = renderer.create(subject);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
