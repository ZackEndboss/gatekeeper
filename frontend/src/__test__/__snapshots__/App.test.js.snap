// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render group1 1`] = `
<div>
  <Header
    links="/"
  />
  <Toolbar />
  <EditServiceDialog />
  <Group
    group="group1"
  />
</div>
`;

exports[`App component render with non existing group 1`] = `
<div>
  <Header
    links="/"
  />
  <Toolbar />
  <EditServiceDialog />
  <div>
    <h1>
      404
    </h1>
    <h3>
      The group was not found.
    </h3>
  </div>
</div>
`;

exports[`App component render without groupId 1`] = `
<div>
  <Header
    links="/"
  />
  <Toolbar />
  <EditServiceDialog />
  <div>
    <h1>
      404
    </h1>
    <h3>
      The group was not found.
    </h3>
  </div>
</div>
`;

exports[`App component render without state 1`] = `
<div>
  <Header
    links="/"
  />
  <Toolbar />
  <EditServiceDialog />
  <div>
    <h1>
      Loading
    </h1>
  </div>
</div>
`;

exports[`load from state should load the state correctly 1`] = `
<div>
  <Header
    links="/"
  />
  <Toolbar />
  <EditServiceDialog />
  <Group
    group="ftx"
  />
</div>
`;
